dpdispatcher/contexts/local_context.py

```diff
@@ -275,7 +278,10 @@ def download(
                         shutil.rmtree(lfile, ignore_errors=True)
                     elif os.path.isfile(lfile) or os.path.islink(lfile):
                         os.remove(lfile)
-                    shutil.copyfile(rfile, lfile)
+                    if not os.path.islink(rfile):
+                        shutil.move(rfile, lfile)
```

Comment:
It might be a good idea to handle the case where `shutil.move` or `shutil.copyfile` fails. This could be done by wrapping these calls in a try-except block and logging the error for easier debugging. [important]